frontend/src/framework/components/ParameterListFilter/parameterListFilter.tsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+
+import { Parameter } from "@framework/EnsembleParameters";
+import { SmartNodeSelector, SmartNodeSelectorSelection } from "@lib/components/SmartNodeSelector";
+import { TreeDataNode } from "@lib/components/SmartNodeSelector";
+import { resolveClassNames } from "@lib/utils/resolveClassNames";
+
+import { isEqual } from "lodash";
+
+// Icons placed here due to limitation of jest for testing utils (cannot import svg)
+import checkIcon from "./private-assets/check.svg";
+import segmentIcon from "./private-assets/segment.svg";
+import {
+    ParameterParentNodeNames,
+    createTreeDataNodeListFromParameters,
+    getParametersMatchingSelectedNodes,
+} from "./private-utils/smartNodeSelectorUtils";
+
+export type ParameterListFilterProps = {
+    parameters: Parameter[];
+    showTitle?: boolean;
+    onChange?: (filteredParameters: Parameter[]) => void;
+};
+
+export const ParameterListFilter: React.FC<ParameterListFilterProps> = (props: ParameterListFilterProps) => {
+    const smartNodeSelectorId = React.useId();
+    const [selectedTags, setSelectedTags] = React.useState<string[]>([ParameterParentNodeNames.IS_NONCONSTANT]);
+    const [selectedNodes, setSelectedNodes] = React.useState<string[]>([]);
+    const [numberOfMatchingParameters, setNumberOfMatchingParameters] = React.useState<number>(0);
+    const [parameters, setParameters] = React.useState<Parameter[] | null>(null);
+    const [treeDataNodeList, setTreeDataNodeList] = React.useState<TreeDataNode[]>([]);
+
+    let candidateTreeDataNodeList = treeDataNodeList;
+    if (parameters === null || !isEqual(props.parameters, parameters)) {
+        candidateTreeDataNodeList = createTreeDataNodeListFromParameters([...props.parameters], checkIcon, segmentIcon);
+        setParameters(props.parameters);
+        setTreeDataNodeList(candidateTreeDataNodeList);
+    }
+    const computedTreeDataNodeList = candidateTreeDataNodeList;
+
+    // Utilizing useEffect to prevent re-render of parent component during rendering
```

Comment:
Consider adding `...during rendering of this component` in order to make it a bit clearer.

Reply:
ok